pywinauto/unittests/test_application.py

```diff
@@ -270,6 +272,26 @@ def test_connect_path(self):
 
         app_conn.UntitledNotepad.MenuSelect('File->Exit')
 
+    def test_connect_path_timeout(self):
+        """Test that connect_() works with a path with timeout"""
+        app1 = Application()
+        def delayed_launch():
+            time.sleep(2)
+            app1.start(_notepad_exe())
+        thread = Thread(target=delayed_launch)
+        thread.start()
+
+        app_conn = Application()
+        app_conn.connect(path=_notepad_exe(), timeout=3)

```

Comment:
Also need a negative test case (there is no such process, `connect` raises an exception after specified timeout (possible precision is ~0.3 sec).